src/Middleware/WebSockets/src/HandshakeHelpers.cs

```diff
@@ -23,6 +23,18 @@ internal static class HandshakeHelpers
             HeaderNames.SecWebSocketVersion
         };
 
+        // "258EAFA5-E914-47DA-95CA-C5AB0DC85B11"
+        private static ReadOnlySpan<byte> _encodedWebSocketKey => new byte[]
+        {
+            (byte)'2', (byte)'5', (byte)'8', (byte)'E', (byte)'A', (byte)'F', (byte)'A', (byte)'5', (byte)'-',
+            (byte)'E', (byte)'9', (byte)'1', (byte)'4', (byte)'-', (byte)'4', (byte)'7', (byte)'D', (byte)'A',
+            (byte)'-', (byte)'9', (byte)'5', (byte)'C', (byte)'A', (byte)'-', (byte)'C', (byte)'5', (byte)'A',
+            (byte)'B', (byte)'0', (byte)'D', (byte)'C', (byte)'8', (byte)'5', (byte)'B', (byte)'1', (byte)'1'
+        };
+
+        [ThreadStatic]
+        private static SHA1 _algorithm;
```

Comment:
@blowdart Is caching SHA1 ok? I know it's reusable, I'm just wondering if the OS resource usage is ok to keep around.

Reply:
So this ends up allocating one for every threadpool thread and they live for the lifetime of the app? That's not  great.

Reply:
@bartonjs for his thoughts

Reply:
SHA-1 state is 96 bytes.  Even if you assume a couple of layers of indirection and metadata it shouldn't be all that large.

The other main alternative would be a static ConcurrentBag for pooling (or going back to a new one per request).

Reply:
We already pool to ton of stuff like this and the savings seem significant.

Reply:
Pooling this gave a 2x gain, so the layers of indirection seem pretty heavy.

Reply:
What do we want to do here? Go back to the slow path and create a new `SHA1` every request, keep the `ThreadStatic`, or maintain a small pool using something like `ConcurrentBag` and creating a temporary new one if the pool is empty?